specs/protocol/isthmus/exec-engine.md

```diff
@@ -131,6 +131,16 @@ is applicable for scenarios where the actual withdrawals root value is not readi
 
 Withdrawals list in the block body is encoded as an empty RLP list.
 
+## EVM Changes
+
+Similar to the `bn256Pairing` precompile in the [granite hardfork](../granite/exec-engine.md),
+[EIP-2537](https://eips.ethereum.org/EIPS/eip-2537) introduces a BLS
+precompile that short-circuits depending on input size in the EVM.
+
+The `BLS12-381` curve operation precompile reverts if its input is
+larger than `<todo>` bytes. This is the input size that consumes
```

Comment:
See `<todo>` here.

Reply:
how do we get this number?

Reply:
Here's the calculations for those:

- G1 multiplication cost: 12000
- G2 multiplication cost: 22500
- G1 MSM:
  - (k * multiplication_cost * max_discount) // 1000 < 20000000 
  - (k * 12000 * 519) // 1000 < 20000000
  - k < 3211.3, so k <= 3211
  - bytes = 160 * k
  - bytes <= 513760

- G2 MSM:
  - (k * multiplication_cost * max_discount) // 1000 < 20000000 
  - (k * 22500 * 524) // 1000 < 20000000
  - k < 1696.35, so k <= 1696
  - bytes = 288 * k
  - bytes <= 488448

- Pairing check:
  - 32600 * k + 37700 < 20000000
  - k <= 612
  - bytes = 384 * k
  - bytes <= 235008

Reply:
> The bn256Pairing precompile execution has additional validation on its input. The precompile reverts if its input is larger than 112687 bytes. This is the input size that consumes approximately 20 M gas given the latest bn256Pairing gas schedule on L2.

https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/granite/exec-engine.md#bn256pairing-precompile-input-restriction

For consistency, we should follow the same reasoning in the above and cap the size to ~20 million gas. Can we fill in the todo before we merge this?

cc @Inphi @ajsutton for visibility

Reply:
TODO filled in with `235008` maximum input byte size where the gas limit is 20M.

This follows @meyer9 's pairing check math.

```
32600 * k + 37700 < 20000000
k <= 612
bytes = 384 * k
bytes <= 235008
```